app/controllers/access_tokens_controller.rb

```diff
@@ -23,17 +23,29 @@ def create
         user_hash = user_info.parsed_response
         puts user_hash
 
-        #saving data about access and refresh tokens to database
-        @token = Token.new(token_hash)
-        @token.save
-
-        #changing 'access_token' key to 'token' so RestClient gem recognizes it for API calls
-        new_hash = {'token' => token_hash['access_token']}
-        new_hash.merge!(token_hash.except!('access_token'))
-        options = {'credentials' => new_hash}
-        options.merge!(user_hash)
+        user = User.where(:spotify_id => user_hash["id"]).first
+        if user.nil?
+            # creat new user
+            user = User.new({
+                :email => user_hash["email"],
+                :spotify_id => user_hash["id"]
+            })
+            user.save!
+            # create new token associated with the user
+            user.create_token(token_hash)
+            existing_user = user
+        end
```

Comment:
Putting this here as an example of how you can create a token that is associated with a new user.